### PR TITLE
Update SEC1 link in ecp.c

### PR DESCRIPTION
**Pull request by xkqian**
Old link doesn't work any more, update it to one
new link to refer version 2

## Description

It will be one simple update.

### CODE REVIEW

File: library/ecp.c [hunk 1]
```diff
@@ -20,7 +20,7 @@
 /*
  * References:
  *
- * SEC1 http://www.secg.org/index.php?action=secg,docs_secg
+ * SEC1 https://www.secg.org/sec1-v2.pdf
```
Comment (yuhaoth) [inline library/ecp.c:23]: How about change this paragraph like this? And add like for RFC 4492 and RFC 7748?
```suggestion
 * - SEC1: https://www.secg.org/sec1-v2.pdf
```

**Review by yuhaoth — CHANGES_REQUESTED** [review]
Some nitpick comments. Please check if it should be changed

File: library/ecp.c (hunk 1 above)
Comment (xkqian) [reply library/ecp.c:23]: Currently prefer to keep it align with the exist comments. 
Anyway, agree that we should re-org the description here, maybe I will raise another PR to deal with that: add links, change comment paragraph,etc.

**Review by yuhaoth — APPROVED** [review]
LGTM

**Review by yanrayw — APPROVED** [review]
LGTM

**Comment by yanrayw** [comment]
Should we also update the links for `library/ecdsa.c` and `library/ecdh.c`?

**Comment by xkqian** [comment]
> Should we also update the links for `library/ecdsa.c` and `library/ecdh.c`?

Good point! The links in these two files should also be updated. Anyway, cause this PR has been merged, we have to move it to another PR: https://github.com/Mbed-TLS/mbedtls/pull/7429, which maybe solve the issue https://github.com/Mbed-TLS/mbedtls/pull/7418#discussion_r1161517083 together.

**Pull request merged**